### ISSUE TITLE
Assign a variable that holds a plain object with mismatched type

### ISSUE THREAD

**Issue by hkleungai** [microsoft/TypeScript#60946]
### 🔎 Search Terms

"assign variable", "variable assignment", "type error", "mismatched types"

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________

### ⏯ Playground Link

https://tsplay.dev/NDpPxN

### 💻 Code

```ts
let x: Record<number, Record<number, boolean>>;
let y = { a: 1, b: 2 };
//  ^?

// Case 1: Assign a plain object with mismatched type.
// As expected, type error.
{
   // @ts-expect-error
   x = { a: 1, b: 2 };
}

// Case 2: Assign a variable that holds a plain object with mismatched type.
// No error emitted. Seems incorrect.
{
   x = y;
}

```

### 🙁 Actual behavior

Unable to assign a variable that holds a plain object with mismatched type.

### 🙂 Expected behavior

Get errors in both cases.

**Comment by MartinJohns** [comment]
This is working as intended. Excess property checks only happens for object literals, that's why the first assignment is an error. But objects are allowed to have additional properties, so the second assignment works, as long as the objects are compatible.
